assignment7.txt

```diff
@@ -0,0 +1,18 @@
+Sean Kee
+
+1. For ++*p, it is read right to left. For *p++, it first adds, then does the *p. For *++p, it also reads right to left.
+
+2. Yes, left to right or right to left is guarenteed because the compiler uses a sort of "order of operations" in it's own sense. For terms that are the same in terms of precedence, the compiler reads right to left.
+
+3. The advantage of using pointers is that a) You can have a string with an undefined length. b) Pass variables by reference rather than value.
+
+4.1 char a[3] = "abc"; for a string
+4.2 Invalid because you are trying to declare an array with a string, then using a - instead of a =
```

Comment:
This is actually valid! "xyz" is just an array of characters, so the "xyz"[1] is just accessing "y". Then you just subtract 'y' from it to get 0. This is because, if you can recall, a char is just a value mapped to a character!